pkgs/development/compilers/go/go-default-pie.patch

```diff
@@ -0,0 +1,47 @@
+diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
+index fb70047dd0..32c5aff8ea 100644
+--- a/src/cmd/dist/build.go
++++ b/src/cmd/dist/build.go
+@@ -648,6 +648,10 @@ func mustLinkExternal(goos, goarch string, cgoEnabled bool) bool {
+ 	}
+ 
+ 	switch goos {
++	case "linux":
++		if goarch != "ppc64" {
++			return true
++		}
+ 	case "android":
+ 		if goarch != "arm64" {
+ 			return true
+diff --git a/src/internal/platform/supported.go b/src/internal/platform/supported.go
+index f9706a6988..120f1a0757 100644
+--- a/src/internal/platform/supported.go
++++ b/src/internal/platform/supported.go
+@@ -114,6 +114,10 @@ func MustLinkExternal(goos, goarch string, withCgo bool) bool {
+ 	}
+ 
+ 	switch goos {
++	case "linux":
++		if goarch != "ppc64" {
++			return true
++		}
+ 	case "android":
+ 		if goarch != "arm64" {
+ 			return true
+@@ -225,7 +229,6 @@ func InternalLinkPIESupported(goos, goarch string) bool {
+ 	switch goos + "/" + goarch {
+ 	case "android/arm64",
+ 		"darwin/amd64", "darwin/arm64",
+-		"linux/amd64", "linux/arm64", "linux/loong64", "linux/ppc64le",
+ 		"windows/386", "windows/amd64", "windows/arm", "windows/arm64":
```

Comment:
We could alternatively try to teach go to use the right elf interpreter path when the internal linker is used.

`GO_LDSO` can be set at build time, or we could teach this part of elf.go to look it up at runtime if setting at build time is inadequate.

src/cmd/link/internal/ld/elf.go:
```go
		if interpreter == "" && buildcfg.GOOS == runtime.GOOS && buildcfg.GOARCH == runtime.GOARCH && buildcfg.GO_LDSO != "" {
			interpreter = buildcfg.GO_LDSO
		}
```